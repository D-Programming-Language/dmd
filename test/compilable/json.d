// PERMUTE_ARGS:
// REQUIRED_ARGS: -o- -X -Xftest_results/compilable/json.out
// POST_SCRIPT: compilable/extra-files/json-postscript.sh


module json;

import std.stdio : writefln;


static this() {}

static ~this() {}


alias int myInt;
myInt x; // bug 3404

struct Foo(T) { T t; }
class  Bar(int T) { int t = T; }
interface Baz(T...) { T[0] t() const; } // bug 3466

template P(alias T) {}

class Bar2 : Bar!1, Baz!(void, 2, null) {
    this() {}
    ~this() {} // bug 4178

    static foo() {}
    protected abstract Foo!int baz();
}

struct Foo2 {
	Bar2 bar2;
	union U {
		struct {
			short s;
			int i;
		}
		Object o;
	}
}

@trusted myInt bar(uint blah, ref Bar2 foo = null) // bug 4477
{
	void inner(void* v)
	{
		assert(true);
	}
	return -1;
}



