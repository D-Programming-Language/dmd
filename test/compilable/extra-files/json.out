[
{
"kind" : "module",
"members" : [
{
"name" : "X",
"kind" : "struct",
"protection" : "public",
"line" : 5}
,
{
"name" : "Y",
"kind" : "enum",
"protection" : "public",
"line" : 7}
,
{
"name" : "myInt",
"kind" : "alias",
"protection" : "public",
"type" : "int",
"line" : 10}
,
{
"name" : "x",
"kind" : "variable",
"protection" : "public",
"type" : "int",
"originalType" : "myInt",
"line" : 11}
,
{
"name" : "Foo3466(T)",
"kind" : "template",
"protection" : "public",
"line" : 14,
"members" : [
{
"name" : "Foo3466",
"kind" : "struct",
"protection" : "public",
"line" : 14,
"members" : [
{
"name" : "t",
"kind" : "variable",
"type" : "T",
"line" : 14}
]
}
]
}
,
{
"name" : "Bar3466(T)",
"kind" : "template",
"protection" : "public",
"line" : 15,
"members" : [
{
"name" : "Bar3466",
"kind" : "class",
"protection" : "public",
"line" : 15,
"members" : [
{
"name" : "t",
"kind" : "variable",
"type" : "T",
"line" : 15}
]
}
]
}
]
}
]
